x-pack/plugins/endpoint/server/routes/whitelist.ts

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter, RequestHandlerContext, APICaller } from 'kibana/server';
+import { schema } from '@kbn/config-schema';
+import { SearchResponse } from 'elasticsearch';
+import { createHash } from 'crypto';
+
+import lzma from 'lzma-native';
+import { WhitelistRule, WhitelistSet } from '../../common/types';
+import { EndpointAppContext } from '../types';
+
+const whitelistIdx: string = 'whitelist'; // TODO: change this
```

Comment:
If these routes index documents in a data index, does that mean the current Kibana user will need ES perms to write to that index? will that be an issue?

Reply:
Yes, is that not always the case? Is there such thing as a Kibana user with read only access?

Reply:
@alexk307 It's possible to have a user with read-only access, yes. I don't know what the expectations are regarding our users, but I'd think it's reasonable to expect that they will have write access? Otherwise, we'll need to lock down certain actions, like whitelisting and opening/closing alerts. Maybe RBAC comes later...

Reply:
Yeah, we need to clear up those expectations because a lot of our app will rely on write access.